src/core/mainloop/connection.c

```diff
@@ -3293,14 +3296,25 @@ connection_bucket_write_limit(connection_t *conn, time_t now)
  *   that's harder to quantify and harder to keep track of.
  */
 bool
-connection_dir_is_global_write_low(const connection_t *conn, size_t attempt)
+connection_dir_is_global_write_low(const connection_t *conn, size_t attempt,
+                                   const compress_method_t c_method)
 {
   size_t smaller_bucket =
     MIN(token_bucket_rw_get_write(&global_bucket),
         token_bucket_rw_get_write(&global_relayed_bucket));
 
   /* Special case for authorities (directory only). */
   if (authdir_mode_v3(get_options())) {
+    /* If this requests is uncompressed and we are configured to reject those,
+     * indicate that we reached the limit. */
+    if (c_method == NO_METHOD &&
+        dirauth_should_reject_uncompressed_requests()) {
+      return true;
+    }
+    if (c_method != NO_METHOD) {
+      /* Always answer compressed request. */
+      return false;
+    }
```

Comment:
This code is missing in 0.4.3, but present in 0.4.2 and master.

Did you forget to delete it in 0.4.2 and master?